Sources/SwiftDocC/Infrastructure/DocumentationConverter.swift

```diff
@@ -37,6 +38,7 @@ public protocol DocumentationConverterProtocol {
 ///
 /// You can also configure the documentation converter to emit extra metadata such as linkable entities and indexing records
 /// information.
+@available(*, deprecated, message: "This deprecated API will be removed after 6.2 is released")
```

Comment:
I noticed a comment near `ConvertActionConverter.convert.recordProblem` mentions `DocumentationConverter` - we might want to remove or update that:

> These problems only represent unexpected thrown errors and aren't particularly user-facing but because
> `DocumentationConverter.convert(outputConsumer:)` emits them as diagnostics we do the same here.

Reply:
I rephrased this comment in https://github.com/swiftlang/swift-docc/pull/1059/commits/84b0dc0fff8b228a248907d277a07e24705de67b to add a future action to change it while still mentioning that this current behavior came from DocumentationConverter in case the reader wonders why the code does this today.